src/HexagonOffload.cpp

```diff
@@ -259,7 +259,15 @@ class InjectHexagonRpc : public IRMutator {
 #else
         debug(1) << "Hexagon device code module: " << device_code << "\n";
         compile_module_to_llvm_bitcode(device_code, "hex.bc");
-        int result = system("${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so");
+
+        int result = 0;
+        string S64 = "${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so";
+        string S128 = S64 +" -mhvx-double";
+        Target target = hexagon_remote_target;
```

Comment:
This should grab the target from the device_code module, and not copy it:

    if (device_code.target().has_feature(Target::HVX_128)) {
        ...
    }